front/pages/api/w/[wId]/data_sources/[name]/databases/[dId]/tables/index.ts

```diff
@@ -0,0 +1,120 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSource } from "@app/lib/api/data_sources";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { CoreAPI } from "@app/lib/core_api";
+import { isDevelopmentOrDustWorkspace } from "@app/lib/development";
+import logger from "@app/logger/logger";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { ListDatabaseTablesResponseBody } from "@app/pages/api/v1/w/[wId]/data_sources/[name]/databases/[dId]/tables";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ListDatabaseTablesResponseBody>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner || !auth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  const plan = auth.plan();
+  if (!owner || !plan) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  if (!isDevelopmentOrDustWorkspace(owner)) {
```

Comment:
Can we add a builder check otherwise we might forget it once we remove that line

Reply:
there is already a builder check